product/src/routes/__test__/delete-review.test.ts

```diff
@@ -0,0 +1,211 @@
+
+  const userId = new mongoose.Types.ObjectId().toHexString();
+
+  // Create a review
+  await request(app)
+    .post(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send({
+      title: "Good Quality",
+      rating: 4,
+      comment:
+        "Purus semper eget duis at tellus. Ut placerat orci nulla pellentesque dignissim enim.",
+    })
+    .expect(201);
+
+  // Make a delete request
+  await request(app)
+    .delete(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send()
+    .expect(200);
+
+  // Find the deleted product
+  const deletedProduct = await Product.findById(product.id);
+
+  expect(deletedProduct!.reviews!.length).toEqual(0);
+  expect(deletedProduct!.reviews![0]).toEqual(undefined);
+});
+
+it("update new rating and numReviews correctly when recieve new several request", async () => {
+  // Create a product
+  const product = await createProduct();
+
+  const userId = new mongoose.Types.ObjectId().toHexString();
+
+  // Create a 1st review
+  await request(app)
+    .post(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send({
+      title: "Good Quality",
+      rating: 4,
+      comment:
+        "Purus semper eget duis at tellus. Ut placerat orci nulla pellentesque dignissim enim.",
+    })
+    .expect(201);
+
+  // Create a 2nd review
+  await request(app)
+    .post(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin())
+    .send({
+      title: "Love It!",
+      rating: 5,
+      comment: "Elit sed vulputate mi sit amet mauris commodo quis imperdiet.",
+    })
+    .expect(201);
+
+  // Make a delete request on 1st review
+  await request(app)
+    .delete(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send()
+    .expect(200);
+
+  // Find the updated product
+  const updatedProduct = await Product.findById(product.id);
+
+  expect(updatedProduct).toBeDefined();
+  expect(updatedProduct!.rating).toEqual(5);
+  expect(updatedProduct!.numReviews).toEqual(1);
+  expect(updatedProduct!.reviews!.length).toEqual(1);
+  expect(updatedProduct!.reviews![0].title).toEqual("Love It!");
+});
+
+it("publishes an event", async () => {
+  // Create a product
+  const product = await createProduct();
+
+  const userId = new mongoose.Types.ObjectId().toHexString();
+
+  // Create a review
+  await request(app)
+    .post(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send({
+      title: "Good Quality",
+      rating: 4,
+      comment:
+        "Purus semper eget duis at tellus. Ut placerat orci nulla pellentesque dignissim enim.",
+    })
+    .expect(201);
+
+  // Make a delete request
+  await request(app)
+    .delete(`/api/products/${product.id}/reviews`)
+    .set("Cookie", global.signin(userId))
+    .send({})
+    .expect(200);
+
+  expect(natsWrapper.client.publish).toHaveBeenCalledTimes(3);
```

Comment:
need to figure out why it called 3 times instead of 2 or change the method from `toHaveBeenCalledTimes` to `toHaveBeenCalled`